### PR TITLE
Add real name checking for Signoff and some enhancements

### PR DESCRIPTION
**Pull request by williamdes**
- Check that Signed-off-by is a real name 
- Update README.md (document events and payload types)
- Removed hard coded repository name
- Moved the secret to the config file
- Improved code (`True` > `true`)
- More changes

----

- Tested against all our signoff lines found on the main repo.
- Supports weird accents

**Have a look at the comment below !**

### CODE REVIEW

**Comment by williamdes** [comment]
:warning: :rescue_worker_helmet: Before you pull any changes, go to `hooks/github.php` and copy the value of `hook_secret` into a safe place.
Then pull the changes and add `define('GITHUB_HOOK_SECRET', 'secretHere!!');` to `hooks/config.php` with the value of the previously copied secret

**Comment by williamdes** [comment]
# A preview of the message
preview

File: hooks/commits.php [hunk 1]
```diff
@@ -21,6 +22,12 @@
     . $contributing_url
     . ") for more information.";
 
+$message_sob_invalid = "<!-- PMABOT:SOB --><!-- PMABOT:SOB:INVALID -->\n"
+    . "This commit has an invalid Signed-Off-By line, please use"
+    . "your full name for legal compliance reasons.\n"
```
Comment (mauriciofauth) [inline hooks/commits.php]: You forgot to add a space here: `please useyour full name`

Comment (williamdes) [reply hooks/commits.php]: Good catch !
fixed

**Comment by williamdes** [comment]
Hello @ibennetch @nijel can you please review my changes?

Btw I tested the changes on my personal test repo.

**Comment by williamdes** [comment]
> BTW: There is also https://github.com/apps/dco which might be used out of the box instead of implementing own code (not sure how it compares, I use this one for @WeblateOrg).

@nijel I did not find a way to add a pattern to dco bot and it generates false positives when checking the email for users using a different email or using the privacy email settings.

**Comment by nijel** [comment]
Okay, I just wanted to mention alternative :-).

File: hooks/commits.php [hunk 2]
```diff
@@ -5,14 +5,15 @@
...
-define('PMAHOOKS', True);
+define('PMAHOOKS', true);
 
 require_once('./lib/github.php');
 
 github_verify_post();
 
 $contributing_url = 'https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md';
 $guidelines_url = 'https://github.com/phpmyadmin/phpmyadmin/wiki/Developer_guidelines';
+$dco_link = 'https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO#L39';
```
Comment (ibennetch) [inline hooks/commits.php]: Possibly you didn't mean to include the `#L39` in the URL

Comment (williamdes) [reply hooks/commits.php]: @ibennetch I finally removed it :)

**Comment by williamdes** [comment]
@ibennetch Can you please install the PR on the server ?
or @nijel ?

PS: dot not forget to read the instructions https://github.com/phpmyadmin/scripts/pull/19#issuecomment-498842981

**Comment by ibennetch** [comment]
I have completed the installation on the server.

I've even tested it with https://github.com/phpmyadmin/phpmyadmin/pull/15340 and it looks great. Thanks.

**Comment by williamdes** [comment]
Thanks for installing the PR :) 
You will new see checked green on webhooks list on each repo since I implemented 🏓 ping requests (github zen) instead of red cross for 500 errors

**Pull request merged**